be/tool/beck.c

```diff
@@ -2395,11 +2502,27 @@ static int cob_proc(struct scanner *s, struct btype *b,
 			m0_free(ca);
 			continue;
 		}
+
 		if ((format_header_verify(ca->coa_val.b_addr,
 					  M0_FORMAT_TYPE_COB_NSREC) == 0) &&
 		    (m0_format_footer_verify(ca->coa_valdata, false) == 0))
-			qput(s->s_q, (struct action *)ca);
+			if (!dry_run)
+				qput(s->s_q, (struct action *)ca);
+			else {
+				m0_buf_free(&ca->coa_key);
+				m0_free(ca);
+			}
```

Comment:
Though, a pair of braces `{}` can be omitted here, but I think for such a long block, it's better to have a pair of braces.